### ISSUE TITLE
simple-obfs failed to handshake

### ISSUE THREAD

**Issue by luckwang** [shadowsocks/simple-obfs#15]
### Server端：
配置环境： 
bandwagon openvz/ubuntu 16.04/shadowsocks-libev-server 2.6.1/最新编译的obfs-server
运行参数：
    "server": "ip",
    "local_address": "127.0.0.1",
    "server_port": 443,
    "local_port": 1080,
    "password": "password",
    "timeout": 300,
    "method": "chacha20",
    “plugin”: "obfs-server --obfs http"

### Client端：
client: lede r2535/shadowsocks-libev 2.6.1/最新编译的obfs-local
运行参数：
  由luci-app-shadowsocks设置，与server配置保持一致
 "plugin": "obfs-local --obfs http --obfs-host www.baidu.com"

### 问题：
无论使用tls或者http.插件都提示failed to handshake，无法连接.

### log:
 2017-01-11 20:54:15 INFO: plugin "obfs-server --obfs tls" enabled
 2017-01-11 20:54:15 INFO: initializing ciphers... chacha20
 2017-01-11 20:54:15 INFO: tcp port reuse enabled
 2017-01-11 20:54:15 INFO: listening at xxx.xxx.xxx.xxx:33045
 2017-01-11 20:54:15 INFO: running from root user
 2017-01-11 20:54:15 [simple-obfs] INFO: obfuscating enabled
 2017-01-11 20:54:15 [simple-obfs] INFO: tcp port reuse enabled
 2017-01-11 20:54:15 [simple-obfs] INFO: listening at xxx.xxx.xxx.xxx:443
 2017-01-11 20:54:15 [simple-obfs] INFO: running from root user
 2017-01-11 20:54:31 INFO: accept a connection
 2017-01-11 20:54:31 ERROR: failed to handshake with xxx.xxx.xxx.xxx
 2017-01-11 20:54:38 INFO: server_recv close the connection
 2017-01-11 20:54:38 INFO: current server connection: 0
 2017-01-11 20:55:11 INFO: accept a connection
 2017-01-11 20:55:11 INFO: connect to xxx.xxx.xxx.xxx:443
 2017-01-11 20:55:11 INFO: remote connected
 2017-01-11 20:55:12 INFO: remote_recv close the connection
 2017-01-11 20:55:12 INFO: current remote connection: 0
 2017-01-11 20:55:12 INFO: current server connection: 0

**Comment by luckwang** [comment]
升级obfs0.0.2之后变成了这样
 2017-01-17 20:44:28 INFO: plugin "obfs-server" enabled
 2017-01-17 20:44:28 INFO: initializing ciphers... chacha20
 2017-01-17 20:44:28 INFO: tcp port reuse enabled
 2017-01-17 20:44:28 INFO: tcp server listening at 127.0.0.1:51739
 2017-01-17 20:44:28 INFO: running from root user
 2017-01-17 20:44:28 [simple-obfs] INFO: tcp port reuse enabled
 2017-01-17 20:44:28 [simple-obfs] INFO: listening at xxx.xxx.xxx.xxx:443
 2017-01-17 20:44:28 [simple-obfs] INFO: running from root user
 2017-01-17 20:44:30 [simple-obfs] ERROR: getpeername: Transport endpoint is not connected
 2017-01-17 20:44:30 [simple-obfs] ERROR: getpeername: Transport endpoint is not connected
 2017-01-17 20:44:31 [simple-obfs] ERROR: getpeername: Transport endpoint is not connected
 2017-01-17 20:44:31 [simple-obfs] ERROR: getpeername: Transport endpoint is not connected
 2017-01-17 20:44:32 [simple-obfs] ERROR: getpeername: Transport endpoint is not connected
 2017-01-17 20:44:32 [simple-obfs] ERROR: getpeername: Transport endpoint is not connected

**Comment by xxxifan** [comment]
same here

**Comment by pexcn** [comment]
请尝试用最新版 shadowsocks-libev 配合使用

**Comment by yukeiyang** [comment]
same

**Comment by LiR1cky** [comment]
same issue as I reported in shadowsocks-android:
https://github.com/shadowsocks/shadowsocks-android/issues/1155

the problem is in the plugin mode with shadowsocks-libev it is always listening to 127.0.0.1 but not the public server ip. We need a parameter in the JSON file (/etc/shadowsocks-libev/config.json) to control this. I tried "server", "server_host", "local_address", but none of them worked. Any suggestion? Or maybe a new commit is required to enable this feature?

Thanks and Regards,
Ricky
